webbpsf/webbpsf_core.py

```diff
@@ -920,6 +921,87 @@ def load_was_opd(self, inputWasOpd, size=1024, save=False, filename='new_was_opd
 
         return fits.HDUList(fits.ImageHDU(newopd, header=hdr))
 
+    def psf_grid(self, all_detectors=True, num_psfs=16, single_psf_centered=True, use_detsampled_psf=False,
+                 save=False, outfile=None, overwrite=True, **kwargs):
+        """
+        Create a PSF library in the form of a grid of PSFs across the detector based on the specified instrument,
+        filter, and detector. The output file is of the form [i, y, x] where i is the PSF position on the detector
+        grid and (y,x) is the 2D PSF.
+
+        Parameters
+        ----------
+        all_detectors: bool
```

Comment:
minor formatting quirk: Annoyingly, the numpy docstring standard specifies that there must be a space _before_ the colon. Like so: 
```suggestion
        all_detectors : bool
```
and likewise for all the other parameters.